lib/parallel_tests.rb

```diff
@@ -20,12 +20,23 @@ def determine_number_of_processes(count)
       ].detect{|c| not c.to_s.strip.empty? }.to_i
     end
 
+    def with_pid_file
+      Tempfile.open('parallel_tests-pidfile') do |f|
+        ENV['PARALLEL_PID_FILE'] = f.path
+        @pids = nil
+        # Pids object should be created before threads will start adding pids to it
+        # Otherwise we would have to use Mutex to prevent creation of several instances
+        @pids = pids
+        yield
+      end
+    end
+
     def pids
       @pids ||= Pids.new(pid_file_path)
     end
 
     def pid_file_path
-      ENV['PARALLEL_PID_FILE'] ||= Tempfile.new('parallel-tests-pidfile').path
+      ENV['PARALLEL_PID_FILE']
```

Comment:
should use `.fetch` so it blow up when it's not there

Reply:
Thanks, it's a very useful idea :)